### ISSUE TITLE
Exporting a gt table to html with spaces in column names produces invalid html headers attributes

### ISSUE THREAD

**Issue by Elias1004** [rstudio/gt#2085]
## Prework

* [x] Read and agree to the [code of conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html) and [contributing guidelines](https://github.com/rstudio/gt/blob/master/.github/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/rstudio/gt/issues), whether open or closed, comment on the existing thread instead of posting a new issue.

## Description

Exporting a gt table to html with spaces in column names produces invalid html header attributes. The generated `header` attributes for `td` tags are not correctly coded and therefore reference non-existing `id`s.

## Reproducible example

```R
library(gt) # Version 1.1.0

sep_character <- "_"

# Data frame with spaces in columns
df <- data.frame(
  `married_low income` = c(30, 2),
  `married_high income` = c(30, 4),
  `single_low income` = c(20, 1),
  `single_high income` = c(20, 2),
  col = c("age", "hsize"),
  check.names = F
)

table <- gt(df, rowname_col = "col") |> tab_spanner_delim(delim = sep_character)
html <- table |> as_raw_html(inline_css = F)

writeLines(html, "gt.html")
```

## Output

```html
<div id="fvixvesmif" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
  <style>
  <!-- ommited -->
</style>
  <table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings gt_spanner_row">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="2" colspan="1" scope="col" id="a::stub"></th>
      <th class="gt_center gt_columns_top_border gt_column_spanner_outer" rowspan="1" colspan="2" scope="colgroup" id="spanner-married_low income">
        <div class="gt_column_spanner">married</div>
      </th>
      <th class="gt_center gt_columns_top_border gt_column_spanner_outer" rowspan="1" colspan="2" scope="colgroup" id="spanner-single_low income">
        <div class="gt_column_spanner">single</div>
      </th>
    </tr>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="married_low-income">low income</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="married_high-income">high income</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="single_low-income">low income</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="single_high-income">high income</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><th id="stub_1_1" scope="row" class="gt_row gt_left gt_stub">age</th>
<td headers="stub_1_1 married_low income" class="gt_row gt_right">30</td>
<td headers="stub_1_1 married_high income" class="gt_row gt_right">30</td>
<td headers="stub_1_1 single_low income" class="gt_row gt_right">20</td>
<td headers="stub_1_1 single_high income" class="gt_row gt_right">20</td></tr>
    <tr><th id="stub_1_2" scope="row" class="gt_row gt_left gt_stub">hsize</th>
<td headers="stub_1_2 married_low income" class="gt_row gt_right">2</td>
<td headers="stub_1_2 married_high income" class="gt_row gt_right">4</td>
<td headers="stub_1_2 single_low income" class="gt_row gt_right">1</td>
<td headers="stub_1_2 single_high income" class="gt_row gt_right">2</td></tr>
  </tbody>
  
</table>
</div>
```

The produced table correctly renders to this:

<img width="444" height="163" alt="Image" src="https://github.com/user-attachments/assets/1d0db881-5008-4999-a0e8-9f54f2fb8609" />

But there are some problems concerning accessibility:

In line 17 the column "married_low income" gets coded to id="married_low-income". But in line 25 the element is referenced incorrectly by `headers="married_low income"`. The header attributes for the other columns are also incorrecly coded.

## Expected result

The columns should be coded to id and header attributes exactly the same way. E.g. `id=married_low-income` and `headers="married_low-income"`

Thank you for your help!

@olhaau @ysaidani

**Comment by ysaidani** [comment]
Hi, I think a solution would be to `make.names()` on column names at some point. 

One option with few downstream implications would be in the HTML export. But at that point, column ids might be referenced in various elements of the `gt_tbl`, and replacing all occasions seems error-prone.

Another, minimally invasive option is doing so early on - upon creation of the `gt_tbl` object would be most convenient and straightforward. Something like `data_tbl <- validate_tbl_colnames(data_tbl)` in `dt_data_init`, right at the end, which would make names syntactically valid and attach original column names as a label attribute. 

```r
#' Check if a tibble has syntactically valid column names, generate them if
#' required and attach original column names as label attribute
#'
#' @param data_tbl
#' @returns The input, potentially modified
#'
#' @noRd
validate_tbl_colnames <- function(data_tbl) {
  orig_colnames <- colnames(data_tbl)
  valid_colnames <- make.unique(make.names(orig_colnames))

  if (!all(orig_colnames == valid_colnames)) {
    # Enfore syntactically valid colnames
    colnames(data_tbl) <- valid_colnames

    # Attach original names as "label" attribute for each column
    for (i in seq_along(data_tbl)) {
      attr(data_tbl[[i]], "label") <- orig_colnames[i]
    }
  }

  data_tbl
}
```

However, I noticed that once labels are set, `tab_spanner_delim()` does not seem to overwrite them, which is undesirable. I wonder though whether `tab_spanner_delim()` should not be operate on column labels rather than column names - which would be fitting, since its output is another set of column labels.

Perhaps somebody with more knowledge of the internals could figure out if there are any downsides to the latter option: enforcing syntactically valid column names upon creation of the `gt_tbl`, setting existing names as labels, and letting `tab_spanner_delim()` operate on column labels rather than names if available.
